### PR TITLE
added call of gpu predictor in new interfaces

### PR DESCRIPTION
**Pull request by makart19**

### CODE REVIEW

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_classification_dense_dpc.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    auto pinterop_model =
+        static_cast<df_interop::interop_model_impl<Task, cls_model_p>*>(model_pimpl.get());
+
+    daal_input.set(daal::algorithms::classifier::prediction::model, pinterop_model->get_model());
+
+    auto daal_voting_mode = df_interop::convert_to_daal_voting_mode(desc.get_voting_mode());
+
+    auto daal_parameter = cls::prediction::Parameter(desc.get_class_count(), daal_voting_mode);
+
+    //daal::data_management::SyclHomogenNumericTable<T>
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_classification_dense_dpc.cpp:83]: delete

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp [hunk 2]
```diff
@@ -53,7 +53,10 @@ TEST(infer_and_train_cls_kernels_test, can_process_simple_case_default_params) {
     const auto y_train_table =
         dal::homogen_table{ queue, y_train, row_count_train, 1, dal::empty_delete<const float>() };
 
-    const auto x_test_table = dal::homogen_table{ x_test_host,
+    auto x_test = sycl::malloc_shared<float>(row_count_test * column_count, queue);
+    queue.memcpy(x_test, x_test_host, sizeof(float) * row_count_test * column_count).wait();
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp:57]: If I am right, `queue.memcpy()` shall be used only with USM pointers. In that case to copy data from host pointer to the USM one, `std::memcpy()` shall be used.

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp [hunk 3]
```diff
@@ -53,7 +53,10 @@ TEST(infer_and_train_cls_kernels_test, can_process_simple_case_default_params) {
...
+    const auto x_test_table = dal::homogen_table{ queue,
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp:58]: use `homogen_table::wrap()`

File: examples/oneapi/dpc/source/decision_forest/df_cls_dense_batch.cpp [hunk 4]
```diff
@@ -33,7 +33,7 @@ void run(sycl::queue &queue) {
   const auto x_train = dal::read<dal::table>(queue, dal::csv::data_source{train_data_file_name});
   const auto y_train = dal::read<dal::table>(queue, dal::csv::data_source{train_label_file_name});
 
-  const auto x_test = dal::read<dal::table>(dal::csv::data_source{test_data_file_name});
+  const auto x_test = dal::read<dal::table>(queue, dal::csv::data_source{test_data_file_name});
   const auto y_test = dal::read<dal::table>(dal::csv::data_source{test_label_file_name});
```
Comment (michael-smirnov) [inline examples/oneapi/dpc/source/decision_forest/df_cls_dense_batch.cpp:37]: queue?

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_classification_dense_dpc.cpp (hunk 1 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_classification_dense_dpc.cpp]: fixed, thanks

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp (hunk 2 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp]: fixed, thanks

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp (hunk 3 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp]: fixed, thanks

File: examples/oneapi/dpc/source/decision_forest/df_cls_dense_batch.cpp (hunk 4 above)
Comment (makart19) [reply examples/oneapi/dpc/source/decision_forest/df_cls_dense_batch.cpp:37]: y_test is used for displaying expected values, therefore no need for queue for it

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp [hunk 5]
```diff
@@ -41,22 +41,17 @@ TEST(infer_and_train_cls_kernels_test, can_process_simple_case_default_params) {
...
-    const auto y_train_table =
-        dal::homogen_table{ queue, y_train, row_count_train, 1, dal::empty_delete<const float>() };
+    std::memcpy(y_train, y_train_host, sizeof(float) * row_count_train);
+    const auto y_train_table = dal::homogen_table::wrap(queue, y_train, row_count_train, 1);
 
-    const auto x_test_table = dal::homogen_table{ x_test_host,
-                                                  row_count_test,
-                                                  column_count,
-                                                  dal::empty_delete<const float>() };
+    auto x_test = sycl::malloc_shared<float>(row_count_test * column_count, queue);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp:52]: Malloc result check?

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp [hunk 6]
```diff
@@ -174,22 +166,17 @@ TEST(infer_and_train_cls_kernels_test, can_process_corner_case) {
     auto queue = sycl::queue(selector);
 
     auto x_train = sycl::malloc_shared<float>(row_count_train * column_count, queue);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp:168]: Malloc result check?

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp [hunk 7]
```diff
@@ -51,22 +51,17 @@ TEST(infer_and_train_reg_kernels_test, can_process_simple_case_default_params) {
     auto queue = sycl::queue(selector);
 
     auto x_train = sycl::malloc_shared<float>(row_count_train * column_count, queue);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp:53]: Malloc result check?

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp [hunk 8]
```diff
@@ -51,22 +51,17 @@ TEST(infer_and_train_reg_kernels_test, can_process_simple_case_default_params) {
...
-    const auto x_train_table = dal::homogen_table{ queue,
-                                                   x_train,
-                                                   row_count_train,
-                                                   column_count,
-                                                   dal::empty_delete<const float>() };
+    std::memcpy(x_train, x_train_host, sizeof(float) * row_count_train * column_count);
+    const auto x_train_table =
+        dal::homogen_table::wrap(queue, x_train, row_count_train, column_count);
 
     auto y_train = sycl::malloc_shared<float>(row_count_train, queue);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp:58]: Malloc result check?

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp [hunk 9]
```diff
@@ -115,22 +111,17 @@ TEST(infer_and_train_reg_kernels_test, can_process_simple_case_non_default_param
...
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp:118]: Malloc result check?

**Review by michael-smirnov — APPROVED** [review]

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp (hunk 5 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp:54]: added

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp (hunk 6 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp:174]: added

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp (hunk 7 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp:53]: added

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp (hunk 8 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp:59]: added

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp (hunk 9 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_reg_kernel_dpc_test.cpp:122]: added

**Comment by makart19** [comment]
@Mergifyio backport rls/2021-gold-mnt

**Pull request merged**